org.eclipse.scout.rt.platform/src/main/java/org/eclipse/scout/rt/platform/html/HtmlHelper.java

```diff
@@ -272,4 +273,147 @@ else if (c == '>') {
+      char nextChar = i < size - 1 ? html.charAt(i + 1) : '\0';
+      if (!inComment && character == '<' && i + 4 < size && "<!--".equals(html.substring(i, i + 4))) {
+        // Comment found <!--
+        inComment = true;
+        result.append(character);
+        continue;
+      }
+      if (inComment && "-->".equals(html.substring(i - 2, i + 1))) {
+        // Closing comment found -->
+        inComment = false;
+        result.append(character);
+        continue;
+      }
+      if (!inComment && !inTag && character == '<') {
+        inTag = true;
+        if (nextChar == '/') {
+          // Closing tag found, e.g. </div>
+          numOpenTags = Math.max(numOpenTags - 1, 0);
+        }
+        else {
+          // Opening tag found e.g. <div>
+          numOpenTags++;
+        }
+        result.append(character);
+        continue;
+      }
+      if (inTag && !inAttr && character == '>') {
+        inTag = false;
+        if (prevChar == '/') {
+          // Self-closing tag found, e.g. <br/>
+          numOpenTags = Math.max(numOpenTags - 1, 0);
+        }
+        result.append(character);
+        continue;
+      }
+      if (inTag && !inAttr && ObjectUtility.isOneOf(character, '"', '\'')) {
+        // Attribute found with " or '
+        inAttr = true;
+        attrChar = character;
+        result.append(character);
+        continue;
+      }
+      if (inAttr && character == attrChar) {
+        // Closing attribute found
+        inAttr = false;
+        result.append(character);
+        continue;
+      }
+      if (!inComment && !inEntity && !inAttr && character == '&') {
+        // HTML entity (character reference) found, e.g. &gt; -> Counts as one character
+        inEntity = true;
+        result.append(character);
+        continue;
+      }
+      if (inEntity && character == ';') {
+        // Closing HTML entity found
+        inEntity = false;
+        numChars++;
+        result.append(character);
+        // Update ellipsis position because an entity counts as character
+        ellipsisPos = updateEllipsisPos(result, maxLength, addEllipsis, ellipsisPos, numChars);
+        continue;
+      }
+      if (inTag || inEntity || inComment) {
+        result.append(character);
+        continue;
+      }
+      numChars++;
+      if (numChars <= maxLength) {
+        result.append(character);
+        if (!inTag && !inEntity) {
+          ellipsisPos = updateEllipsisPos(result, maxLength, addEllipsis, ellipsisPos, numChars);
+        }
+      }
+      else {
+        addEllipsis(result, ellipsisPos);
+        ellipsisPos = -1;
+      }
+    }
+    return result.toString();
+  }
+
+  private int updateEllipsisPos(StringBuilder result, int maxLength, boolean addEllipsis, int ellipsisPos, int numChars) {
+    if (addEllipsis && ellipsisPos == -1 && numChars == maxLength) {
+      ellipsisPos = result.length();
+    }
+    return ellipsisPos;
+  }
+
+  private void addEllipsis(StringBuilder text, int position) {
```

Comment:
Since the name "addEllipsis" is already used for the parameter (which is not involved in this method), maybe "insertEllipsis" would be more appropriate?

```suggestion
  private void insertEllipsis(StringBuilder text, int position) {
```